### ISSUE TITLE
Fix MacOS builds

### ISSUE THREAD

**Issue by ahockersten** [openliero/openliero#13]
MacOS currently does not build. @grawlinson has started working on this.

**Comment by grawlinson** [comment]
Update: I have it working via cli, just not via the normal macOS bundle (`$APP.app`) due to a few factors:

* SDL2/SDL2_image need to be statically linked, bundling the SDL frameworks inside the macOS bundle isn't recommended, and making users responsible for managing dependencies is a no-go for obvious reasons.
* I don't actually know anything about macOS bundles, so that's a whole new avenue that I have to look into.
* The same goes for the current platform-specific code that we have (`sdlmain.cpp`, `main.cpp`, and `SDLMain.{h,m}`)

tl;dr I need more coffee and free time.

**Comment by grawlinson** [comment]
macOS builds work, but bundles (openliero.app) do not. Will require further research.
